File: osis/labs/LR-7-8/block-2/main.pas
var
  s: string;
  i: integer;
  count_plus_minus: integer;
  count_before_zero: integer;

begin
  writeln('Введите строку:');
  readln(s);
  count_plus_minus := 0;
  count_before_zero := 0;

  for i := 1 to length(s) do
  begin
    if (s[i] = '+') or (s[i] = '-') then
    begin
      count_plus_minus := count_plus_minus + 1;
      if (i < length(s)) and (s[i+1] = '0') then
        count_before_zero := count_before_zero + 1;
    end;
  end;

  writeln('Общее количество "+" и "-": ', count_plus_minus);
  writeln('Количество "+" или "-" перед "0": ', count_before_zero);
end.

File: osis/labs/LR-7-8/block-2/main.py
s = input("Введите строку: ")

count_plus_minus = 0
count_before_zero = 0

for i in range(len(s)):
    if s[i] == '+' or s[i] == '-':
        count_plus_minus += 1 
        if i + 1 < len(s) and s[i+1] == '0':
            count_before_zero += 1 
            
            
print("Общее количество '+' и '-':", count_plus_minus)
print("Количество '+' или '-' перед '0':", count_before_zero)
